python/CMakeLists.txt

```diff
@@ -2,28 +2,76 @@
 # The MPACT Compiler Python Modules
 #-------------------------------------------------------------------------------
 
+# Disables generation of "version soname" (i.e. libFoo.so.<version>), which
+# causes pure duplication as part of Python wheels.
+set(CMAKE_PLATFORM_NO_VERSIONED_SONAME ON)
+
+
 # The directory at which the Python import tree begins.
 set(MPACT_PYTHON_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}/mpact")
 
+# We vendor our own MLIR instance in the `torch_mlir` namespace.
+add_compile_definitions("MLIR_PYTHON_PACKAGE_PREFIX=mpact.")
+
 declare_mlir_python_sources(MPACTPythonSources)
+declare_mlir_python_sources(MPACTPythonExtensions)
 
 declare_mlir_python_sources(MPACTPythonSources.PublicAPI
   ROOT_DIR "${MPACT_PYTHON_ROOT_DIR}"
   ADD_TO_PARENT MPACTPythonSources
   SOURCES
     mpactbackend.py
+  )
+
+#-------------------------------------------------------------------------------
+# Extensions
+#-------------------------------------------------------------------------------
+
+declare_mlir_python_extension(MPACTPythonExtensions.Main
+  MODULE_NAME _mpact
+  ADD_TO_PARENT MPACTPythonExtensions
+  SOURCES
+    MPACTModule.cpp
+  EMBED_CAPI_LINK_LIBS
+    MPACTCAPI
+  PRIVATE_LINK_LIBS
+    LLVMSupport
 )
 
 #-------------------------------------------------------------------------------
 # Python Modules
 #-------------------------------------------------------------------------------
 
+# To compiler TorchMLIRPythonExtension (which registers torch dialect and
```

Comment:
do we now package torch-mlir python into mpact?

Reply:
yeah